test/api/route/messages_test.dart

```diff
@@ -3,16 +3,177 @@ import 'dart:convert';
+      bool? applyMarkdown,
+    }) async {
+      final result = await getMessageCompat(connection,
+        messageId: messageId,
+        applyMarkdown: applyMarkdown,
+      );
+      expectLegacy
+        ? (check(connection.lastRequest).isA<http.Request>()
+            ..method.equals('GET')
+            ..url.path.equals('/api/v1/messages')
+            ..url.queryParameters.deepEquals({
+              'narrow': jsonEncode([ApiNarrowMessageId(messageId)]),
+              'anchor': messageId.toString(),
+              'num_before': '0',
+              'num_after': '0',
+              if (applyMarkdown != null) 'apply_markdown': applyMarkdown.toString(),
+              'client_gravatar': 'true',
+            }))
+        : (check(connection.lastRequest).isA<http.Request>()
+            ..method.equals('GET')
+            ..url.path.equals('/api/v1/messages/$messageId')
+            ..url.queryParameters.deepEquals({
+              'message_id': messageId.toString(),
+              if (applyMarkdown != null) 'apply_markdown': applyMarkdown.toString(),
+            }));
+      return result;
+    }
+
+    test('modern; message found', () {
+      return FakeApiConnection.with_((connection) async {
+        final message = eg.streamMessage();
+        final fakeResult = GetMessageResult(message: message);
+        connection.prepare(json: fakeResult.toJson());
+        final result = await checkGetMessageCompat(connection, false,
+          messageId: message.id,
+          applyMarkdown: true,
+        );
+        check(result).isNotNull().jsonEquals(message);
+      });
+    });
+
+    test('modern; message not found', () {
+      return FakeApiConnection.with_((connection) async {
+        final message = eg.streamMessage();
+        final fakeResponseJson = {
+          'code': 'BAD_REQUEST',
+          'msg': 'Invalid message(s)',
+          'result': 'error',
+        };
+        connection.prepare(httpStatus: 400, json: fakeResponseJson);
+        final result = await checkGetMessageCompat(connection, false,
+          messageId: message.id,
+          applyMarkdown: true,
+        );
+        check(result).isNull();
+      });
+    });
+
+    test('legacy; message found', () {
+      return FakeApiConnection.with_((connection) async {
+        final message = eg.streamMessage();
+        final fakeResult = GetMessagesResult(
+          anchor: message.id,
+          foundNewest: false,
+          foundOldest: false,
+          foundAnchor: true,
+          historyLimited: false,
+          messages: [message],
+        );
+        connection.prepare(json: fakeResult.toJson());
+        final result = await checkGetMessageCompat(connection, true,
+          messageId: message.id,
+          applyMarkdown: true,
+        );
+        check(result).isNotNull().jsonEquals(message);
+      }, zulipFeatureLevel: 119);
+    });
+
+    test('legacy; message not found', () {
+      return FakeApiConnection.with_((connection) async {
```

Comment:
nit:
```suggestion
      return FakeApiConnection.with_(zulipFeatureLevel: 119, (connection) async {
```
(instead of at the end of the callback).

Dart didn't support that syntax until something like a year or two ago — previously named arguments had to come after positional arguments — which is a reason you'll often still see code not take advantage of it. But I think a case like this where the named argument is short and important for understanding the code, and the positional one is many lines long, is an excellent use of it.